src/CMakeLists.txt

```diff
@@ -370,6 +370,18 @@ add_library(Halide
             $<TARGET_OBJECTS:Halide_initmod>)
 add_library(Halide::Halide ALIAS Halide)
 
+# TODO: there is probably a more elegant way to do these conditionals in CMake
+if (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
```

Comment:
This should be `AppleClang` because clang might still use the GNU linker on Linux.